application/apps/indexer/dlt/src/dlt_net.rs

```diff
@@ -272,6 +322,92 @@ pub async fn create_index_and_mapping_dlt_from_socket(
     res
 }
 
+pub (crate) struct SessionProcessor {
+    tmp_writer: BufWriter<std::fs::File>,
+    chunk_factory: ChunkFactory,
+    buf_writer: BufWriter<std::fs::File>,
+    line_nr: usize,
+    tag: String,
+    update_channel: cc::Sender<ChunkResults>
+}
+
+impl SessionProcessor {
+    fn new(
+        session_id: String,
+        out_path: &std::path::PathBuf,
+        initial_line_nr: usize,
+        tag: &str,
+        update_channel: cc::Sender<ChunkResults>) -> Result<Self, anyhow::Error> {
```

Comment:
we should not use anyhow::Error anymore in our library, we use `thiserror`
since the sessionprocessor operates in the connection `dlt_net` module, it should use an error that can be converted to `ConnectionError`

Reply:
Added one more type of error:
```
#[derive(Debug, Error)]
pub enum ConnectionError {
    #[error("socket configuration seems to be broken: {}", cause)]
    WrongConfiguration { cause: String },
    #[error("could not establish a connection: {}", reason)]
    UnableToConnect { reason: String },
    #[error("could establish session file: {}", error)]              <<================
    UnableGetSessionFile { error: String },                          <<================
    #[error("error trying to connect: {}", info)]
    Other { info: String },
    #[error(transparent)]
    Any(#[from] anyhow::Error),
}
```
And removed `anyhow::Error`